main.tf

```diff
@@ -0,0 +1,193 @@
+# TODO: (input) Create ACM cert outside of this module
+
+# TODO: use `terraform-terraform-label` to generate all labels.
+
+# TODO: (output) security group IDs
+
+#module "container_definition" {
+#  source    = "git::https://github.com/cloudposse/terraform-aws-ecs-container-definition.git?ref=init"
+#  container_name      = "app"
+#  container_image    = "nginx"
+#}
+
+# ECR repository
+resource "aws_ecr_repository" "app" {
+  name = "${var.app_name}"
+}
+
+# Cloudwatch Log Group
+resource "aws_cloudwatch_log_group" "app" {
+  name = "${var.app_name}"
+
+  tags {
+    Stage = "${var.stage}"
+    Application = "${var.app_name}"
+  }
+}
+
+# ECS Task def
+data "template_file" "web_task" {
+  template = "${file("task_definition.json")}"
+
+  vars {
+    image     = "${aws_ecr_repository.app.repository_url}"
+    log_group = "${aws_cloudwatch_log_group.app.name}"
+  }
+}
+
+resource "aws_ecs_task_definition" "web" {
+  family                = "${var.stage}_web"
+  container_definitions = "${data.template_file.web_task.rendered}"
+
+  #container_definitions = "${module.container_definition.container_definitions}"
+  requires_compatibilities = ["FARGATE"]
+  network_mode             = "awsvpc"
+  cpu                      = "${var.task_cpu}"
+  memory                   = "${var.task_memory}"
+  execution_role_arn       = "${aws_iam_role.ecs_execution_role.arn}"
+  task_role_arn            = "${aws_iam_role.ecs_execution_role.arn}"
+}
+
+# ALB
+
+## Target Group
+resource "random_id" "target_group_suffix" {
+  byte_length = 2
+}
+
+resource "aws_alb_target_group" "alb_target_group" {
+  name        = "${var.stage}-alb-target-group-${random_id.target_group_suffix.hex}"
```

Comment:
Move to `label`

Reply:
hopefully label is sufficiently disambiguated that we do not need the random id.

Reply:
I think we'd still need something here to make this unique - I mainly put this in with blue/green style deploys in mind (see below lifecycle rule).